main.go

```diff
@@ -114,31 +112,21 @@ func (c *Controller) handleNewCSR(key string) error {
 		return nil
 	}
 
-	approvalMsg := "This CSR was approved by the Node CSR Approver"
-	machineList, err := c.machineClient.MachineV1beta1().Machines(machineAPINamespace).List(metav1.ListOptions{})
-	if err == nil {
-		err := authorizeCSR(machineList, csr, parsedCSR)
-		if err != nil {
-			// Don't deny since it might be someone else's CSR
-			glog.Infof("CSR %s not authorized: %v", csr.GetName(), err)
-			return err
-		}
-	}
+	machines, err := c.machineClient.MachineV1beta1().Machines(machineAPINamespace).List(metav1.ListOptions{})
 	if err != nil {
-		glog.Infof("machine api not available: %v", err)
-		// Validate the CSR for the bootstrapping phase without a SAN check.
-		_, err := validateCSRContents(csr, parsedCSR)
-		if err != nil {
-			glog.Infof("CSR %s not valid: %v", csr.GetName(), err)
-			return err
-		}
-		approvalMsg += " (no SAN validation)"
+		return fmt.Errorf("failed to list machines: %v", err)
```

Comment:
Do we want to do this now? testing this PR would then be contingent on both the bootstrap self-approve script and PR to disable the auto-approve. just wondering

Reply:
If it passes CI we know that we don't need the failover, right?

Reply:
Also, you could get a transient error and hit the failover path because the code is too blunt.  I think we need to remove it before GA so might as well do it now.

Reply:
> If it passes CI we know that we don't need the failover, right?

Yeah, otherwise removing it might stop us from bootstrapping with no machine API available. So let's see what CI says.. I am +1 on getting rid of the failover here though.